swap/src/bitcoin/wallet.rs

```diff
@@ -194,6 +190,12 @@ impl Wallet {
             .ok_or_else(|| anyhow!("Could not get raw tx with id: {}", txid))
     }
 
+    pub async fn status_of_script(&self, script: &Script, txid: &Txid) -> Result<ScriptStatus> {
+        let mut client = self.client.lock().await;
+        client.subscribe_to_script(script.clone())?;
```

Comment:
It doesn't make any sense to subscribe here. The lock is still active so the state cannot possible change between this line and you asking for the status in the next one.

Also, a function named `status_of_script` should return you the status and not have any side-effects!